src/app/Providers/components/VMwareProviderHostsTable/VMwareProviderHostsTable.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import { Alert, Button, Level, LevelItem, Pagination } from '@patternfly/react-core';
+import { Table, TableHeader, TableBody, sortable, ICell, IRow } from '@patternfly/react-table';
+import { usePaginationState, useSortState } from '@app/common/hooks';
+import { useSelectionState } from '@konveyor/lib-ui';
+import { IVMwareProvider, IHost } from '@app/queries/types';
+import { formatHostNetwork } from './helpers';
+import SelectNetworkModal from './SelectNetworkModal';
+import { useHostsQuery, useProvidersQuery } from '@app/queries';
+import LoadingEmptyState from '@app/common/components/LoadingEmptyState';
+
+interface IVMwareProviderHostsTableProps {
+  providerId?: string;
+}
+
+const VMwareProviderHostsTable: React.FunctionComponent<IVMwareProviderHostsTableProps> = ({
+  providerId,
+}: IVMwareProviderHostsTableProps) => {
+  const hostsQuery = useHostsQuery(providerId);
+  const providersQuery = useProvidersQuery();
+
+  const hosts: IHost[] = !hostsQuery.data ? [] : hostsQuery.data;
+
+  const columns: ICell[] = [
+    { title: 'Name', transforms: [sortable] },
+    { title: 'Network for migration data transfer', transforms: [sortable] },
+    { title: 'Bandwidth', transforms: [sortable] },
+    { title: 'MTU', transforms: [sortable] },
+  ];
+
+  const getSortValues = (host: IHost) => {
+    const { name, network, bandwidth, mtu } = host;
+    // First cell is the generated checkbox from using onSelect.
+    return ['', name, 'network', 'bandwidth', 'mtu'];
+    //TODO update when api data is available
+    // return ['', name, formatHostNetwork(network), bandwidth, mtu];
+  };
+  const { sortBy, onSort, sortedItems } = useSortState(hosts, getSortValues);
+  const { currentPageItems, setPageNumber, paginationProps } = usePaginationState(sortedItems, 10);
+  const { selectedItems, toggleItemSelected, selectAll } = useSelectionState<IHost>({
+    items: sortedItems,
+  });
+
+  const rows: IRow[] = sortedItems.map((host: IHost) => {
+    const { name } = host;
+    return {
+      meta: { host },
+      selected: selectedItems.includes(host),
+      cells: [name, 'network N/A', 'bandwidth N/A', 'mtu N/A'],
+      //TODO: update when api info is available
+      // cells: [name, formatHostNetwork(host.metadata.network), bandwidth, mtu],
+    };
+  });
+
+  const [isSelectNetworkModalOpen, setIsSelectNetworkModalOpen] = React.useState(false);
+
+  return (
+    <>
+      {hostsQuery.isLoading || providersQuery.isLoading ? (
+        <LoadingEmptyState />
+      ) : hostsQuery.isError ||
+        providersQuery.isError ||
+        hostsQuery?.data?.length === 0 ||
```

Comment:
Isn't is possible to have a provider with no hosts? That wouldn't be an error, we'd just have the "no hosts" empty state appear here in that case right?

Reply:
@ibolton336 if it makes the logic easier you could even lift the `useHostsQuery` up into the HostsPage and handle all the loading states / errors etc there, and only render the table component if there are defined hosts